### DISCUSSION TITLE
Show model *physical size* in selection drop-down

### DISCUSSION THREAD

**Discussion by Mugane** [open-webui/open-webui#7924] [Ideas]
# Feature Request

Although the model size (parameters) is shown in the chat UI drop-down, the physical size is not (unlike in the settings page where you can view the parameters and RAM requirements). The RAM is arguably more important in making a speed determination, and since it's quite tedious to stop a slow model  after it has been selected/run, it is probably very much worth including this data in the drop-down view. Pic below of what it looks like currently, with redundant parameter information and missing RAM requirements. 

![image](https://github.com/user-attachments/assets/5a66ac5b-92d0-4af3-9aa1-d8a402a3a021)

It would be really helpful to use the same kind of display as the model select on the admin page that shows the RAM:

![image](https://github.com/user-attachments/assets/f3d0849e-6b94-415a-be8f-62c0710013a6)

**Comment by Mugane** [comment]
If someone could just give me a little direction on where this change should be made I'll write the implementation

**Comment by Mugane** [comment]
Hello?

**Comment by frenzybiscuit** [comment]
Slow model: anything 70B and above.

Fast model: 32B and below

Seems easy enough?

**Reply by Mugane** [reply]
Hardly. All the models have different sizes and different systems have different VRAM capacities. The only way to know if a model will fit or not is to have the size displayed.

**Reply by frenzybiscuit** [reply]
> Hardly. All the models have different sizes and different systems have different VRAM capacities. The only way to know if a model will fit or not is to have the size displayed.

Physical size is irrelevant.

size + context size + context quant (q4/q6/q8/fp16) = vram usage.

**Reply by Mugane** [reply]
Are you actually suggesting that people run `f()=size + context size + context quant (q4/q6/q8/fp16)` on each model in their heads as they choose one from the front-end ux drop-down?
As far as local models are concerned, size is literally the MOST IMPORTANT FACTOR when it comes to inference speed. if it fits in VRAM, it is going to be fast. If it doesn't, it isn't. The size is probably somewhat linearly proportional to speed (load time, run time) within these constraints. But even if it isn't (I'm not interested in arguments about logs or minor exponents), it's CORRELATED. So if you have 24GB VRAM you can likely safely expect it to run fast models up to that capacity. Maybe leave 10-20% for context etc - that is not even remotely the point of this issue.
If you can do a calculation of actual use then go for it. Until then, just display the stupid size of the model so that users can at least see the lower limit of their ram use. This seems so trivial, why are we arguing about minutia?

**Reply by frenzybiscuit** [reply]
No dude/gal im saying model size is irrelevant.

I could run a 1.5B model @ 2GB size or whatever and use 48GB VRAM if the model supports a high enough context size.

like the qwen 1M 7B model.

what I’m saying is this feature request is pointless.

**Reply by Mugane** [reply]
When I say "physical size" what do you think I am saying? I am talking about the literal space used in memory.

**Reply by frenzybiscuit** [reply]
> When I say "physical size" what do you think I am saying? I am talking about the literal space used in memory.

Oh, here I was thinking you meant the physical size of the model.

great, prs are welcome. Make sure your calculations include the memory used by context.

**Reply by Mugane** [reply]
Oh thank goodness I thought I was going mad and had a fundamental misunderstanding lol. any idea where the file/template is for the model selection or is there tech documentation for the ux? happy to write/test a display adjustment but it's a big project to go on a hunt...

**Comment by tjbck** [comment]
![image](https://github.com/user-attachments/assets/79b1ecea-319d-49e6-9907-a591cb09c168)

It's already displayed.

**Reply by Mugane** [reply]
I'm not seeing that at all. What version are you using and what browser? Thx

**Reply by Mugane** [reply]
![Screenshot from 2025-04-08 04-25-30](https://github.com/user-attachments/assets/1b5433f2-181d-4203-8449-f3587771503a)

**Reply by Mugane** [reply]
Ohhhhh you have to hover over the *model size* to show the physical size. Thanks

It would be much more useful to show the physical size without having to hover. This hover over each one, one-by-one, is clumsy.

**Reply by Mugane** [reply]
In fact, why not have a table where you can just sort by size or name or quant size or physical size - the things that one might consider when selecting a model...

**Comment by Mugane** [comment]
@tjbck Why would you decree that it is not a relevant discussion lol? Clearly there is at least an issue of clarity in the UX... You know where it is shown because... you created it... but seems like nobody else does...
